pkg/client/cache/reflector.go

```diff
@@ -285,7 +287,11 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			return nil
 		}
 		if err := r.watchHandler(w, &resourceVersion, resyncCh, stopCh); err != nil {
-			if err != errorResyncRequested && err != errorStopRequested {
+			if err == errorResyncRequested {
+				glog.V(1).Infof("%s: watch of %v ending because handler requested resync", r.name, r.expectedType)
+			} else if err == errorStopRequested {
+				glog.V(1).Infof("%s: watch of %v ending because handler requested stop", r.name, r.expectedType)
+			} else {
```

Comment:
I can see an argument for taking this down to V(2) or lower.  But we have a few paths here which do a full refetch without any logging, and I think we need some logging on all the paths, so that we can start to better understand the flakes we're seeing.